app/src/main/java/de/Maxr1998/xposed/maxlock/ui/LockActivity.java

```diff
@@ -107,20 +156,32 @@ public void onWindowFocusChanged(boolean hasFocus) {
     @Override
     protected void onStart() {
         super.onStart();
-        GoogleAnalytics.getInstance(this).reportActivityStart(this);
+        //GoogleAnalytics.getInstance(this).reportActivityStart(this);
     }
 
     @Override
     public void onStop() {
         super.onStop();
-        GoogleAnalytics.getInstance(this).reportActivityStop(this);
+        /*GoogleAnalytics.getInstance(this).reportActivityStop(this);*/
         if (PreferenceManager.getDefaultSharedPreferences(this).getBoolean(Common.ENABLE_PRO, false) &&
                 PreferenceManager.getDefaultSharedPreferences(this).getBoolean(Common.ENABLE_LOGGING, false) && !unlocked) {
             Util.logFailedAuthentication(this, requestPkg);
         }
+        //Stop spying on users, i know it's just google analytics but, the same time you help google collecting datas on every one
+        //This app definitely does not need internet connection
         if (!isInFocus) {
             Log.d("MaxLock/LockActivity", "Lost focus, finishing.");
+            //forceFinishML();
             finish();
         }
     }
+
+    public void forceFinishML() {
+        //FUCK IT AT THE END NOT WHILE FLIRTING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        ActivityManager am = (ActivityManager) getSystemService(Activity.ACTIVITY_SERVICE);
+        am = (ActivityManager) getSystemService(Activity.ACTIVITY_SERVICE);
+        am.killBackgroundProcesses("de.Maxr1998.xposed.maxlock");
+        //FUCK IT AT THE END NOT WHILE FLIRTING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        //INTIKA
+    }
```

Comment:
Hahaha :) 
yeah i was a little mad in that comment i agree, i lost 2 days debugging because of that kill in the middle of nowhere 

i don't use forcefinish because it's verry dangerous for the rest of the app and it need deep debug before putting any kill any where

i am not using it because i did not yet implemented "force kill hook" feature that will be using it

Comment updated :)